internal/user/repository/view/user_session_view.go

```diff
@@ -1,123 +1,76 @@
 package view
 
 import (
+	"context"
+	"database/sql"
+	_ "embed"
+	"errors"
+
 	"github.com/jinzhu/gorm"
+	"github.com/zitadel/logging"
 
-	"github.com/zitadel/zitadel/internal/domain"
 	usr_model "github.com/zitadel/zitadel/internal/user/model"
 	"github.com/zitadel/zitadel/internal/user/repository/view/model"
 	"github.com/zitadel/zitadel/internal/view/repository"
 	"github.com/zitadel/zitadel/internal/zerrors"
 )
 
-func UserSessionByIDs(db *gorm.DB, table, agentID, userID, instanceID string) (*model.UserSessionView, error) {
-	userSession := new(model.UserSessionView)
-	userAgentQuery := model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyUserAgentID,
-		Method: domain.SearchMethodEquals,
-		Value:  agentID,
-	}
-	userQuery := model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyUserID,
-		Method: domain.SearchMethodEquals,
-		Value:  userID,
-	}
-	instanceIDQuery := &model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyInstanceID,
-		Method: domain.SearchMethodEquals,
-		Value:  instanceID,
-	}
-	query := repository.PrepareGetByQuery(table, userAgentQuery, userQuery, instanceIDQuery)
-	err := query(db, userSession)
-	if zerrors.IsNotFound(err) {
-		return nil, zerrors.ThrowNotFound(nil, "VIEW-NGBs1", "Errors.UserSession.NotFound")
-	}
-	return userSession, err
-}
+//go:embed user_session_by_id.sql
+var userSessionByIDQuery string
 
-func UserSessionsByUserID(db *gorm.DB, table, userID, instanceID string) ([]*model.UserSessionView, error) {
-	userSessions := make([]*model.UserSessionView, 0)
-	userAgentQuery := &usr_model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyUserID,
-		Method: domain.SearchMethodEquals,
-		Value:  userID,
-	}
-	instanceIDQuery := &usr_model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyInstanceID,
-		Method: domain.SearchMethodEquals,
-		Value:  instanceID,
-	}
-	query := repository.PrepareSearchQuery(table, model.UserSessionSearchRequest{
-		Queries: []*usr_model.UserSessionSearchQuery{userAgentQuery, instanceIDQuery},
-	})
-	_, err := query(db, &userSessions)
-	return userSessions, err
-}
+//go:embed user_sessions_by_user_agent.sql
+var userSessionsByUserAgentQuery string
 
-func UserSessionsByAgentID(db *gorm.DB, table, agentID, instanceID string) ([]*model.UserSessionView, error) {
-	userSessions := make([]*model.UserSessionView, 0)
-	userAgentQuery := &usr_model.UserSessionSearchQuery{
-		Key:    usr_model.UserSessionSearchKeyUserAgentID,
-		Method: domain.SearchMethodEquals,
-		Value:  agentID,
+func UserSessionByIDs(db *gorm.DB, agentID, userID, instanceID string) (*model.UserSessionView, error) {
+	tx, err := db.DB().BeginTx(context.Background(), &sql.TxOptions{ReadOnly: true})
```

Comment:
Why does the function still takes a `*gorm.DB` if only the underlying DB is required? Can't we use our own `database.DB` and its tx wrappers instead?